### PR TITLE
Maria Pogosjans To-do 

### PR DESCRIPTION
**Pull request by MariaPogosjan**
Link to my deployed project:
https://pogggis-todo.netlify.app/

My reflections on how this week's project turned out:
Finally, we got the chance to do our own to-do application :)  I am addicted to my post-its and now I can use my own app for it.  

Things I'd like to have clarified or explained in more detail:
I would like to understand more about dispatch and when to use the redux store and regular useStates.

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+import { Container } from './components/Container'
+
+const reducer = combineReducers({
+  todos: todos.reducer
+})
+
+const persistedState = loadState()
+const store = createStore(reducer, persistedState)
+
+store.subscribe(() => {
```
Comment (emmabartl) [inline code/src/App.js:16]: Cool that you use the subscribe method🤓

File: code/src/components/Container.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  if(items.length === 0) {
+    return(
+      <main>
+        <StartHeader />
+        <lottie-player
+          src='https://assets3.lottiefiles.com/packages/lf20_jy1bgnpp.json' 
+          speed='1'
+          loop
+          autoplay>
+      </lottie-player>
```
Comment (emmabartl) [inline code/src/components/Container.js:22]: Cool animation! It's moving very fast in my browser, not sure if it's my computer or if its supposed to be more chill👀

File: code/src/components/Filter.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      setFilter(items.filter((todo) => todo.isComplete))
+        break;
+      case 'uncompleted': 
+      setFilter(items.filter((todo) => !todo.isComplete))
+        break; 
+      default:
+        setFilter(items)
+        break;
+    }
+  }
```
Comment (emmabartl) [inline code/src/components/Filter.js:28]: Nice usage of switch!

File: code/src/components/TodoItem.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import Checkbox from '@material-ui/core/Checkbox'
+import DeleteIcon from '@material-ui/icons/Delete'
+import EditIcon from '@material-ui/icons/Edit'
+import SaveIcon from '@material-ui/icons/Save'
+
+import todos from '../reducers/todos'
+
+const TodoItem = ({ checked, id, text, handleEditTodos }) => {
```
Comment (emmabartl) [inline code/src/components/TodoItem.js:10]: Love this component and it's functionality 🤩

File: code/src/index.css [hunk 5]
```diff
@@ -1,13 +1,116 @@
...
+.todo-container {
+  width: 80%;
+  flex-wrap: wrap;
+}
+
+.todo-header {
+  margin-bottom: -10px;
+}
+
+
```
Comment (emmabartl) [inline code/src/index.css:47]: Clean up lines🧹

File: code/src/reducers/localStorage.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+export const saveState = (state) => {
+  try {
+    const serializedState = JSON.stringify(state);
+    localStorage.setItem('state', serializedState);
+  } catch {
+    // ignore write errors
+  }
+};
```
Comment (emmabartl) [inline code/src/reducers/localStorage.js:21]: Cool solution to save to local storage🙌

**Review by emmabartl — APPROVED** [review]
Maria! I'm so impressed of what you've accomplished in this project👑  It looks so nice! 
I love how you structure your code. It looks so clean and is very easy to follow👏 I really like that you made it possible for the users to update their tasks and making them able to reset tasks. That's a supercool feature! Great job🥳

**Review by KarolCislowski — APPROVED** [review]
Nice app 🥇

**Pull request closed without merging**